privacy-config/privacy-config-impl/src/main/java/com/duckduckgo/privacy/config/impl/observers/LocalPrivacyConfigObserver.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.privacy.config.impl.observers
+
+import android.content.Context
+import androidx.annotation.WorkerThread
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import com.duckduckgo.app.global.DispatcherProvider
+import com.duckduckgo.di.scopes.AppObjectGraph
+import com.duckduckgo.privacy.config.impl.PrivacyConfigPersister
+import com.duckduckgo.privacy.config.impl.R
+import com.duckduckgo.privacy.config.impl.di.PrivacyConfigCoroutineScope
+import com.duckduckgo.privacy.config.impl.models.JsonPrivacyConfig
+import com.duckduckgo.privacy.config.impl.network.JSONObjectAdapter
+import com.squareup.anvil.annotations.ContributesMultibinding
+import com.squareup.moshi.Moshi
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@WorkerThread
+@Singleton
+@ContributesMultibinding(AppObjectGraph::class)
+class LocalPrivacyConfigObserver @Inject constructor(
+    private val context: Context,
+    private val privacyConfigPersister: PrivacyConfigPersister,
+    @PrivacyConfigCoroutineScope val coroutineScope: CoroutineScope,
```

Comment:
I think we can safely remove the `PrivacyConfigCoroutineScope` here. I don't see any `couroutineScope` provider dep that is annotated with it. So presumably we're just getting the `applicationCoroutineScope`
Maybe what you really want is to annotate with `@AppCoroutineScope` here?